Neos.Flow/composer.json

```diff
@@ -44,11 +44,11 @@
         "ramsey/uuid": "^3.0 || ^4.0",
         "laminas/laminas-code": "^4.10",
 
-        "doctrine/orm": "^2.12.0",
-        "doctrine/migrations": "^3.0",
-        "doctrine/dbal": "^2.13",
-        "doctrine/common": "^3.0.3",
-        "doctrine/annotations": "^1.12",
+        "doctrine/orm": "^2.13.0",
+        "doctrine/migrations": "^3.3",
+        "doctrine/dbal": "^3.3",
```

Comment:
(Why) does it have to be specific 3.3+ can't we have
```suggestion
        "doctrine/dbal": "^3",
```

Reply:
That goes back to https://github.com/neos/flow-development-collection/pull/2637/commits/a3d3b7ed3a57aecb4f3e278fa7e5841f22330a42 so I have no clue, we could try it I guess 🤷

Reply:
Makes it easier for Composer to filter possible versions. `^3` is `>= 3.0.0, <4.0.0`, but `^3.3` is `>=3.3.0, < 4.0.0`, so it makes solving dependencies easier…

Reply:
I get that, but are we really not compatible with 3.2 for example?

Reply:
Quick test on the site, when I change it locally to ^3 and do `composer update  --prefer-lowest` it will sitll install 3.3.0, tl;dr; it doesn't matter. 3.3 seems the minimum our network of dependencies needs now.

Reply:
y the way, 3.0 is insecure, 3.1 at least partly:
<img width="272" alt="image" src="https://github.com/neos/flow-development-collection/assets/95873/21d12e6f-11bd-488d-8e72-c16dbe4c4ee3">

Reply:
> it doesn't matter. 3.3 seems the minimum our network of dependencies needs now.

But it _does_ matter IMO. At some point we might be able to update `doctrine/orm` for example, those requirements might change so IMO we should follow the specify the lowest and highest version that the package itself supports.
In _general_ that is of course. We have to weight against the work that means ofc.
And in this specific case it's probably not an issue because 3rd party packages that support 3.0 will hopefully also support 3.3

> By the way, 3.0 is insecure, 3.1 at least partly

That ought to be covered by `roave/security-advisories`

Reply:
> That ought to be covered by `roave/security-advisories`

Yes, but you can still help the resolver. Why waste time and energy? Just because Composer 2 is a ton faster than Composer 1, this is not irrelevant.

Reply:
That being said, it could be:
```suggestion
        "doctrine/dbal": "^3.1.4",
```

Reply:
yep, that was my point, sry for the confusion :)

Reply:
alright, alright, will do :)

Reply:
done